lib/kilter.rb

```diff
@@ -149,19 +149,17 @@ def track(col)
 	## performed on all tracked fields.
 	#############################################################################
 	def count(col = nil)
-		count_operation = col.nil? || col.empty?
+		count_operation = col.blank?
 
 		return self unless (@tracked.keys.include?(col) || count_operation)
 		return @tracked[col] unless count_operation
 
-		@filtered_rset.each do |r|
-			@tracked.keys.each do |k|
-				# use send to handle columns that may be overriden in models
-				value = r.send(k.to_s)
-				@tracked[k][value.to_s] += 1 unless (value.nil? || value.try(:empty?))
+		@tracked.keys.each do |k|
+			groups = @filtered_rset.model.from(@filtered_rset).group(k.to_s).count
```

Comment:
There is a method called model which may not be used in this code - if not, we can probably delete it.